bin/lib/zopen-build

```diff
@@ -1588,6 +1590,126 @@ LABEL specification=1.0.0"
   cd -
 }
 
+generateZOSLIBHooks() 
+{
+printHeader "Generating ZOSLIB Environment Hooks"
+zoslib_env="PATH|prepend|PROJECT_ROOT/bin
+LIBPATH|prepend|PROJECT_ROOT/lib
+"
+
+if command -V "${ZOPEN_APPEND_TO_ZOSLIB_ENV_CODE}" >/dev/null 2>&1; then
+  printVerbose "Appending additional environment zoslib variables..."
+  zoslib_env="${zoslib_env}$(${ZOPEN_APPEND_TO_ZOSLIB_ENV_CODE})"
+fi
+
+# Generate directories
+mkdir -p .zoslib_hooks
+cd .zoslib_hooks
+
+output="zoslib_env_hook.c"
+
+# Defines zoslib_env_hook
+
+cat <<zz >$output
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+__attribute__((visibility("default"))) int zoslib_env_hook(char*) __attribute__((used));
+
+__attribute__((visibility("default"))) int zoslib_env_hook(char* root_dir) {
+char* envar_value;
+char* value_str;
+char* pos;
+long size;
+zz
+
+
+# Process input
+echo "$zoslib_env" | while read line; do
+  var=$(echo $line | cut -d '|' -f 1)
+  action=$(echo $line | cut -d '|' -f 2)
+  value=$(echo $line | cut -d '|' -f 3)
+
+  if [ "$action" = "unset" ]; then
+    echo "unsetenv(\"$var\");" >> $output
+  elif [ "$action" = "set" ]; then
+    cat <<zz >>$output
+    value_str = "$value";
+    size = PATH_MAX;
+    envar_value = (char*)malloc(size);
+    memset(envar_value, 0, size);
+
+    // Substitute PROJECT_ROOT with root_dir
+    pos = strstr(value_str, "PROJECT_ROOT");
+    if (pos != NULL) {
+      int len = pos - value_str;
+      strncat(envar_value, value_str, len);
+      strcat(envar_value, root_dir);
+      strcat(envar_value, pos + strlen("PROJECT_ROOT"));
```

Comment:
since PROJECT_ROOT is a hard-coded string, why not have a macro with that string and then you can have a PROJECT_ROOT_LEN that is just `sizeof(PROJECT_ROOT)-1 `? That saves a strlen call...